extension/test/unit/submit-payment-details.handler.spec.js

```diff
@@ -0,0 +1,144 @@
+    expect(request.browserInfo).to.deep.equal(submitPaymentDetailsRequest.browserInfo)
+    expect(request.amount).to.deep.equal(submitPaymentDetailsRequest.amount)
+    expect(request.merchantAccount).to.equal(process.env.ADYEN_MERCHANT_ACCOUNT)
+
+    const setCustomFieldAction = response.actions.find(a => a.action === 'setCustomField')
+    expect(setCustomFieldAction.name).to.equal('submitAdditionalPaymentDetailsResponse')
+    expect(setCustomFieldAction.value).to.be.a('string')
+    expect(setCustomFieldAction.value).to.equal(addInterfaceInteraction.fields.response)
+
+    const addTransaction = response.actions.find(a => a.action === 'addTransaction')
+    expect(addTransaction.transaction).to.be.a('object')
+    expect(addTransaction.transaction.type).to.equal('Authorization')
+    expect(addTransaction.transaction.state).to.equal('Success')
+    expect(addTransaction.transaction.interactionId)
+      .to.equal(JSON.parse(submitPaymentDetailsSuccessResponse).pspReference)
+  })
+
+  it('when resultCode from Adyen is "ChallengeShopper", '
+    + 'then it should return actions "addInterfaceInteraction" and "setCustomField"',
+    async () => {
+      scope.post('/payments/details')
+        .reply(200, submitPaymentDetailsChallengeResponse)
+
+      const ctpPaymentClone = _.cloneDeep(ctpPayment)
+      ctpPaymentClone.custom.fields.submitAdditionalPaymentDetailsRequest = JSON.stringify(submitPaymentDetailsRequest)
+      ctpPaymentClone.custom.fields.makePaymentResponse = JSON.stringify(makePaymentRedirectResponse)
+
+      const response = await execute(ctpPaymentClone)
+
+      expect(response.actions).to.have.lengthOf(2)
+
+      const addInterfaceInteraction = response.actions.find(a => a.action === 'addInterfaceInteraction')
+      expect(addInterfaceInteraction.fields.type).to.equal('submitPaymentDetails')
+      expect(addInterfaceInteraction.fields.request).to.be.a('string')
+      expect(addInterfaceInteraction.fields.response).to.be.a('string')
+      expect(addInterfaceInteraction.fields.createdAt).to.be.a('string')
+
+      const request = JSON.parse(addInterfaceInteraction.fields.request)
+      expect(request.reference).to.deep.equal(submitPaymentDetailsRequest.reference)
+      expect(request.riskData).to.deep.equal(submitPaymentDetailsRequest.riskData)
+      expect(request.paymentMethod).to.deep.equal(submitPaymentDetailsRequest.paymentMethod)
+      expect(request.browserInfo).to.deep.equal(submitPaymentDetailsRequest.browserInfo)
+      expect(request.amount).to.deep.equal(submitPaymentDetailsRequest.amount)
+      expect(request.merchantAccount).to.equal(process.env.ADYEN_MERCHANT_ACCOUNT)
+
+      const setCustomFieldAction = response.actions.find(a => a.action === 'setCustomField')
+      expect(setCustomFieldAction.name).to.equal('submitAdditionalPaymentDetailsResponse')
+      expect(setCustomFieldAction.value).to.be.a('string')
```

Comment:
Actually here when I check the custom field has a resultCode with ChallangeShopper and this is the idea of all this test, I would really consider that type of checks, it is not a random string(to.be.a('string')) so it needs to be checked.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/271/commits/b7e712a0625a8011180d587ec90a7a0eae4e97c8